lib/service/domain_service.dart

```diff
@@ -9,55 +13,63 @@ abstract class DomainService {
 }
 
 class DomainServiceImpl implements DomainService {
-  static const String _tnsDomain = 'https://api.tezos.domains/graphql';
-  static const String _ensDomain =
-      'https://gateway-arbitrum.network.thegraph.com/api/'
-      '780901e32f46e70908727c94d3119788/subgraphs/id/'
-      '5XqPmWe6gjyrJtFn9cLy237i4cWw2j9HcUJEXsP5qGtH';
-  static const String _tnsQuery = '''
-    { domains(where: { name: { in: ["<var>"] } }) { items { address name} } }
-  ''';
-
-  static const String _ensQuery = '''
-    { domains(where: {name: "<var>"}) { name resolvedAddress { id } } }
+  static final String _addressEndpoint = Environment.domainResolverUrl;
+
+  static const String _addressQuery = '''
+    query {
+      lookup(inputs: [
+        { chain: "<chain>", name: "<var>", skipCache: false },
+      ]) {
+        chain
+        name
+        address
+        error
+      }
+    }
   ''';
 
-  static GraphClient get _tnsClient => GraphClient(_tnsDomain);
+  static const String _subKey = 'lookup';
 
-  static GraphClient get _ensClient => GraphClient(_ensDomain);
+  static GraphClient get _ensClient => GraphClient(_addressEndpoint);
 
-  @override
-  Future<String?> getEthAddress(String domain) async {
+  Future<String?> _getAddress(String domain, String chain) async {
     try {
       final result = await _ensClient.query(
-        doc: _ensQuery.replaceFirst('<var>', domain),
-        subKey: 'domains',
-      );
+        doc: _addressQuery
+            .replaceFirst('<var>', domain)
+            .replaceFirst('<chain>', chain),
+        subKey: _subKey,
+      ) as List?;
       if (result == null || result.isEmpty) {
         return null;
       }
-      return result.first['resolvedAddress']['id'];
+      return result.first['address'];
     } catch (e) {
       return null;
     }
   }
 
+  @override
+  Future<String?> getEthAddress(String domain) async {
+    try {
+      final ethAddress = await _getAddress(domain, 'ethereum');
+      return ethAddress;
+    } catch (e, s) {
+      unawaited(Sentry.captureException(
```

Comment:
it is normal for user to type a domain that has not register with ens.
especially when he has not finish typing.
so i don't think we should add Sentry here